server/schedulers/utils.go

```diff
@@ -149,3 +151,166 @@ const (
 func newTaintCache(ctx context.Context) *cache.TTLUint64 {
 	return cache.NewIDTTL(ctx, taintCacheGCInterval, taintCacheTTL)
 }
+
+// ScoreInfo stores storeID and record of a store.
+type ScoreInfo struct {
+	storeID uint64
+	score   float64
+}
+
+// NewScoreInfo returns a ScoreInfo.
+func NewScoreInfo(storeID uint64, score float64) *ScoreInfo {
+	return &ScoreInfo{
+		storeID: storeID,
+		score:   score,
+	}
+}
+
+// GetStoreID returns the storeID.
+func (s *ScoreInfo) GetStoreID() uint64 {
+	return s.storeID
+}
+
+// GetScore returns the score.
+func (s *ScoreInfo) GetScore() float64 {
+	return s.score
+}
+
+// SetScore sets the score.
+func (s *ScoreInfo) SetScore(score float64) {
+	s.score = score
+}
+
+// ScoreInfos is used for sorting ScoreInfo.
+type ScoreInfos struct {
+	scoreInfos []*ScoreInfo
+	isSorted   bool
+}
+
+// NewScoreInfos returns a ScoreInfos.
+func NewScoreInfos() *ScoreInfos {
+	return &ScoreInfos{
+		scoreInfos: make([]*ScoreInfo, 0),
+	}
+}
+
+// Add adds a scoreInfo into the slice.
+func (s *ScoreInfos) Add(scoreInfo *ScoreInfo) {
+	s.scoreInfos = append(s.scoreInfos, scoreInfo)
+}
+
+// Len returns length of slice.
+func (s *ScoreInfos) Len() int { return len(s.scoreInfos) }
+
+// Less returns if one number is less than another.
+func (s *ScoreInfos) Less(i, j int) bool { return s.scoreInfos[i].score < s.scoreInfos[j].score }
+
+// Swap switches out two numbers in slice.
+func (s *ScoreInfos) Swap(i, j int) {
+	s.scoreInfos[i], s.scoreInfos[j] = s.scoreInfos[j], s.scoreInfos[i]
+}
+
+// Sort sorts the slice.
+func (s *ScoreInfos) Sort() {
+	sort.Sort(s)
+	s.isSorted = true
+}
+
+// GetScoreInfo returns the scoreInfos.
+func (s *ScoreInfos) GetScoreInfo() []*ScoreInfo {
+	return s.scoreInfos
+}
+
+// GetMin returns the min of the slice.
+func (s *ScoreInfos) GetMin() *ScoreInfo {
+	if !s.isSorted {
```

Comment:
Why do we need it?

Reply:
It's used in `NormalizeStoresStats`.

Reply:
But before we use `GetMin`, it seems we will call `Sort` first

Reply:
Good, I forgot it